torch/_refs/__init__.py

```diff
@@ -704,6 +713,24 @@ def sub(
     aten_op=None,  # CompositeImplicitAutograd
 )
 
+
+def _xlogy(a: Union[Tensor, NumberType], b: Union[Tensor, NumberType]):
+    if isinstance(a, Tensor) and isinstance(b, Number):
+        b = prims._wrap_scalar(b, dtype=a.dtype, device=a.device)
+    elif isinstance(b, Tensor) and isinstance(a, Number):
+        a = prims._wrap_scalar(a, dtype=b.dtype, device=b.device)
+
+    rhs = where(eq(a, 0), a, mul(a, log(b)))
```

Comment:
This is a perfect implementation

Do you think that doing `where(eq(a, 0), 0, mul(a, log(b)))` might be faster for trace executors like nvFuser in the future? Because maybe they could insert the constant zero without having to read `a`?

Reply:
Reading a constant zero should benefit nvFuser.